Source/Core/S2Cell.js

```diff
@@ -0,0 +1,665 @@
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.number("index", index);
+  if (index < 0 || index > 3) {
+    throw new DeveloperError("child index must be in the range [0-3].");
+  }
+  if (this._level === 30) {
+    throw new DeveloperError("cannot get child of leaf cell.");
+  }
+  //>>includeEnd('debug');
+
+  var newLsb = lsb(this._cellId) >> BigInt(2); // eslint-disable-line
+  var childCellId = this._cellId + BigInt(2 * index + 1 - 4) * newLsb; // eslint-disable-line
+  return new S2Cell(childCellId);
+};
+
+/**
+ * Gets the parent cell of an S2Cell.
+ *
+ * @returns {S2Cell} Returns the parent of the S2Cell.
+ * @private
+ */
+S2Cell.prototype.getParent = function () {
+  //>>includeStart('debug', pragmas.debug);
+  if (this._level === 0) {
+    throw new DeveloperError("cannot get parent of root cell.");
+  }
+  //>>includeEnd('debug');
+  var newLsb = lsb(this._cellId) << BigInt(2); // eslint-disable-line
+  return new S2Cell((this._cellId & (~newLsb + BigInt(1))) | newLsb); // eslint-disable-line
+};
+
+/**
+ * Gets the parent cell at the given level.
+ *
+ * @returns {S2Cell} Returns the parent of the S2Cell.
+ * @private
+ */
+S2Cell.prototype.getParentAtLevel = function (level) {
+  //>>includeStart('debug', pragmas.debug);
+  if (this._level === 0 || level < 0 || this._level < level) {
+    throw new DeveloperError("cannot get parent at invalid level.");
+  }
+  //>>includeEnd('debug');
+  var newLsb = lsbForLevel(level);
+  return new S2Cell((this._cellId & -newLsb) | newLsb);
+};
+
+/**
+ * @private
+ */
+S2Cell.fromFacePosLevel = function (face, pos, level) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.bigint("pos", pos);
+  if (face < 0 || face > 5) {
+    throw new DeveloperError("Invalid S2 Face (must be within 0-5)");
+  }
+
+  if (level < 0 || level > S2MaxLevel) {
+    throw new DeveloperError("Invalid S2 Face (must be within 0-5)");
+  }
+  if (pos < 0 || pos > Math.pow(4, level)) {
+    throw new DeveloperError("Invalid Hilbert position for level");
+  }
+  //>>includeEnd('debug');
+  var cell = new S2Cell((face << BigInt(S2PositionBits)) + (pos | BigInt(1))); // eslint-disable-line
+  return cell.getParentAtLevel(level);
+};
+
+/**
+ * Get center of the S2 cell.
+ *
+ * @param {Ellipsoid} [options.ellipsoid=Ellipsoid.WGS84] The ellipsoid.
+ * @returns {Cartesian} The position of center of the S2 cell.
+ * @private
+ */
+S2Cell.prototype.getCenter = function (ellipsoid) {
+  ellipsoid = defaultValue(ellipsoid, Ellipsoid.WGS84);
+
+  var center = getS2Center(this._cellId);
+  // Normalize XYZ.
+  center = Cartesian3.normalize(center, center);
+  var cartographic = new Cartographic.fromCartesian(
+    center,
+    Ellipsoid.UNIT_SPHERE
+  );
+  // Interpret as geodetic coordinates on the ellipsoid.
+  return Cartographic.toCartesian(cartographic, ellipsoid, new Cartesian3());
+};
+
+/**
+ * Get vertex of the S2 cell.
+ *
+ * @param {Number} index An integer index of the vertex.
```

Comment:
Maybe add a note that this is an integer in `[0, 3]` and it wraps CCW around the cell.